src/config.rs

```diff
@@ -24,23 +24,13 @@ use uuid::Uuid;
 
 mod builder;
 mod config_type;
-mod endpoints;
 mod error;
-mod metadata;
-
-pub(crate) use self::{
-    error::ValueInvalidArgs,
-    metadata::{extract_endpoint_tokens, parse_endpoint_metadata_from_yaml},
-};
-
-pub use self::{
-    builder::Builder,
-    config_type::ConfigType,
-    endpoints::{
-        EmptyListError, Endpoints, RetainedItems, UpstreamEndpoints, UpstreamEndpointsIter,
-    },
-    error::ValidationError,
-};
+
+use crate::endpoint::Endpoint;
```

Comment:
> if we let it start importing modules from other places we end up with a cyclic references

This isn't a thing in Rust though. In Rust the module system is separate from imports, you can reference anything from anywhere and it has no impact on the outputted code. In general you want more general types to be higher in the module tree as child modules can reference private items from parents. Which is why this is in its own module, so that it has the same level visibility to the sibling modules `config`, `filters`, and `cluster`.

Reply:
I didn't mean that the compiler won't be able to handle it but rather that we will infact be creating cycles if A->importsB->..importsA which is best avoided. I guess its not as weird if they're sibling modules but still feels weird to me having config import endpoint :man_shrugging: